lib/rules/no-return-await.js

```diff
@@ -44,7 +46,32 @@ module.exports = {
             context.report({
                 node: context.getSourceCode().getFirstToken(node),
                 loc: node.loc,
-                messageId: "redundantUseOfAwait"
+                messageId: "redundantUseOfAwait",
+                suggest: [
+                    {
+                        messageId: "removeAwait",
+                        fix(fixer) {
+                            const areAwaitAndAwaitedExpressionOnTheSameLine = node.loc.start.line === node.argument.loc.start.line;
```

Comment:
We can safely fix the code if the `argument` is parenthesized and the opening parenthesis is on the same line as `await`.

For example:

```js
async () => {
    return await (
        foo()
    )
};

```

In this example, however, `node` and `node.argument` do not start on the same line, so per the current condition the suggestion will not be provided.

We could do something like this:

```js
const awaitToken = sourceCode.getFirstToken(node);
const tokenAfterAwait = sourceCode.getTokenAfter(awaitToken);
const areAwaitAndAwaitedExpressionOnTheSameLine = awaitToken.loc.start.line === tokenAfterAwait.loc.start.line;
```

(`sourceCode` is now needed in multiple places, so we could declare `const sourceCode = context.getSourceCode();` in `create(context)`).

Reply:
I'm happy to implement this, though I would actually recommend not to do it, as it increases the complexity, risks of potentially introducing errors when running the suggestion if we missed a special case, and because suggestion from my point of view don't need to cover all use cases.
I'm a bit torn, though, as this might be a relevant case in a situation like the following:
```ts
async () => {
  return await (
    await fetch("https://really-long-domain.com/with-an-even-longer-path")
  ).json()
}
```

Reply:
It's fine to implement this, it isn't overly complex and we have similar logic in other rules.

Reply:
It seems that `getTokenAfter` is not getting the token immediately after the note:
1. For `async () => await bar()`, it seems that `sourceCode.getTokenAfter(awaitToken)` is `null`.
2. For
   ```ts
   async function foo() {
     return await new Promise(resolve => {
       resolve(5);
     });
   }
   ```
   `getTokenAfter` gives the semicolon 2 lines after the `await`, which leads to missing a suggestion that could happen.
   
Instead, this worked:
```ts
const [awaitToken, tokenAfterAwait] = sourceCode.getTokens(node);
```

Reply:
> * For `async () => await bar()`, it seems that `sourceCode.getTokenAfter(awaitToken)` is `null`.
> * For
>   ```ts
>   async function foo() {
>     return await new Promise(resolve => {
>       resolve(5);
>     });
>   }
>   ```
>   
>   
>       
>         
>       
>   
>         
>       
>   
>       
>     
>   `getTokenAfter` gives the semicolon 2 lines after the `await`, which leads to missing a suggestion that could happen.

This looks like the result of `sourceCode.getTokenAfter(node)`. Did you try this, it should work:

```js
const awaitToken = sourceCode.getFirstToken(node);
const tokenAfterAwait = sourceCode.getTokenAfter(awaitToken);
```

Reply:
Or this:

```js
const [awaitToken, tokenAfterAwait] = sourceCode.getFirstTokens(node, 2);
```

Reply:
Thanks! Yes, I indeed tried to `getTokenAfter(node)` 🤦
Looking at the alternatives now, I actually feel like the current solution with `getTokens` reads cleanest and would keep it - what do you think?

Reply:
`sourceCode.getTokens(node)` would go from the start till the end of the node to create and return an array with all tokens of the node, while we need only the first two. If you prefer a solution with getting all the tokens we need at once, `sourceCode.getFirstTokens(node, 2)` would be more optimal.

Reply:
Thanks, I'll change it right away

Reply:
Done

Reply:
Good work crew, I appreciate all 👍